src/main/scripts/src/app/shared/permissions/table/permissions-table.component.ts

```diff
@@ -47,54 +52,308 @@ export class PermissionsTableComponent implements OnInit {
+        this.route.paramMap.subscribe((params: ParamMap) => {
+          let orgCode: string = params.get('organizationCode');
+          let proCode: string = params.get('projectCode');
+          switch (this.entityType) {
+            case 'users':
+              for (let r in this.roles) {
+                if (this.rolesCheckbox[this.roles[r]] && this.roles.hasOwnProperty(r)) {
+                  this.addProjectUsersGroupsRole(orgCode, proCode, this.roles[r], [name], []);
+                  addedRoles.push(this.roles[r]);
+                }
+              }
+            break;
+            case 'groups':
+              for (let r in this.roles) {
+                if (this.rolesCheckbox[this.roles[r]] && this.roles.hasOwnProperty(r)) {
+                  this.addProjectUsersGroupsRole(orgCode, proCode, this.roles[r], [], [name]);
+                  addedRoles.push(this.roles[r]);
+                }
+              }
+              break;
+          }
+        });
+        break;
+    }
+    if (addedRoles.length !== 0) {
+      this.entities.push({name: name, roles: addedRoles});
+      this.otherEntities.splice(this.otherEntities.indexOf(name), 1);
+      this.emptyCheckboxes();
+    }
+  }
+
+  public onRemove(index: number) {
+    switch (this.resourceType) {
+      case 'organization':
+        this.route.paramMap.subscribe((params: ParamMap) => {
+          let orgCode: string = params.get('organizationCode');
+          switch (this.entityType) {
+            case 'users':
+              this.entities[index]['roles'].forEach(role => {
+                this.removeOrganizationUsersGroupsRole(orgCode, role, [this.entities[index]['name']], []);
+              });
+              break;
+            case 'groups':
+              this.entities[index]['roles'].forEach(role => {
+                this.removeOrganizationUsersGroupsRole(orgCode, role, [], [this.entities[index]['name']]);
+              });
+              break;
+          }
+        });
+        break;
+      case 'project':
+        this.route.paramMap.subscribe((params: ParamMap) => {
+          let orgCode: string = params.get('organizationCode');
+          let proCode: string = params.get('projectCode');
+          switch (this.entityType) {
+            case 'users':
+              this.entities[index]['roles'].forEach(role => {
+                this.removeProjectUsersGroupsRole(orgCode, proCode, role, [this.entities[index]['name']], []);
+              });
+              break;
+            case 'groups':
+              this.entities[index]['roles'].forEach(role => {
+                this.removeProjectUsersGroupsRole(orgCode, proCode, role, [], [this.entities[index]['name']]);
+              });
+              break;
+          }
+        });
+    }
+    this.otherEntities.push(this.entities[index]['name']);
+    this.entities.splice(index, 1);
+  }
+
+  public changePermission(entityName: string, role: string, event) {
+    let addPermision: boolean = event.target.checked;
```

Comment:
You do not need to put type annotations on variables if you assign values directly. In such cases types are automatically inferred by TypeScript.